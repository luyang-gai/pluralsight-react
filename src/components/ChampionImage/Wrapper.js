import styled from 'styled-components';

const Wrapper = styled.img`
  display: block;
  margin: auto;
`;

export default Wrapper;
