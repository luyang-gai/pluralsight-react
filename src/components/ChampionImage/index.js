/**
 *
 * ChampionImage
 *
 * Component to render a champion image.
 * Input: championId
 * Output: <img> of champion
 */

import React  from 'react';
import PropTypes from 'prop-types';

import ImageHelpers from './ImageHelpers';
import Wrapper from './Wrapper';

const ChampionImage = (props) => {
  return (
    <Wrapper src={ImageHelpers.getImageByChampionId(props.championId)} {...props}/>
  );
};

ChampionImage.propTypes = {
  championId: PropTypes.number.isRequired
};

export default ChampionImage;
