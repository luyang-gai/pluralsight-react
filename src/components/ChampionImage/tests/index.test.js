import React from 'react';
import ChampionImage from '../index';
import renderer from 'react-test-renderer';

it('renders a basic ChampionImage correctly', () => {
  const tree = renderer.create(
    <ChampionImage championId={62}/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
