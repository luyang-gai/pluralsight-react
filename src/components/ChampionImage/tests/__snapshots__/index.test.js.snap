// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a basic ChampionImage correctly 1`] = `
<img
  className="EhGPD"
  src="http://ddragon.leagueoflegends.com/cdn/7.9.1/img/champion/Wukong.png"
/>
`;
